src/main/kotlin/com/sourcegraph/cody/ui/HtmlViewer.kt

```diff
@@ -21,7 +26,16 @@ object HtmlViewer {
                 ChatUIConstants.TEXT_MARGIN,
                 ChatUIConstants.TEXT_MARGIN,
                 ChatUIConstants.TEXT_MARGIN))
-    jEditorPane.addHyperlinkListener(BrowserHyperlinkListener.INSTANCE)
+    jEditorPane.addHyperlinkListener(MyBrowserHyperlinkListener(project))
     return jEditorPane
   }
+
+  class MyBrowserHyperlinkListener(val project: Project) : BrowserHyperlinkListener() {
```

Comment:
Maybe `MyBrowserHyperlinkListener` could be a project level service? 🤔 that would allow us to use it without passing project via `CodyOnboardingGuidancePanel` and `createHtmlViewer`.

Reply:
Or maybe TelemetryV2 should be a project level service. It seems to make sense

Reply:
unfortunately it would not change anything because you would still need provide project for `getInstance`.